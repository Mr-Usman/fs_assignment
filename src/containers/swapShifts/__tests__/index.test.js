import "babel-polyfill";
import React from "react";
import { render, unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";
import Component from "../index";
import { create } from "react-test-renderer";
import { mount } from "enzyme";
import { MemoryRouter } from "react-router-dom";
import toJson from "enzyme-to-json";

/** mock data */
const location = {
  state: {
    user: {
      email: ""
    }
  }
};

jest.mock("axios", () => ({
  post: jest.fn(() => {
    return {
      data: "usman"
    };
  })
}));

describe("<SwapShift /> Component", () => {
  let wrapper;
  it("mounting the component(for unauthorize user)", () => {
    const wrap = mount(
      <MemoryRouter>
        <Component location={location} role="developer" />
      </MemoryRouter>
    );
    expect(toJson(wrap)).toMatchSnapshot();
    wrap.unmount();
  });

  beforeEach(() => {
    wrapper = mount(
      <MemoryRouter>
        <Component location={location} role="manager" />
      </MemoryRouter>
    );
  });

  afterAll(() => {
    wrapper.unmount();
  });

  it("mounting the component", () => {
    // console.log(wrapper.html());
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
