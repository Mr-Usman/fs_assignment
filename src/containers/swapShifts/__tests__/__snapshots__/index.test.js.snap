// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SwapShift /> Component mounting the component 1`] = `
<MemoryRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "cpf0mn",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "cpf0mn",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <SwapShifts
      location={
        Object {
          "state": Object {
            "user": Object {
              "email": "",
            },
          },
        }
      }
      role="manager"
    >
      <withRouter(Navigationbar)
        role="manager"
      >
        <Navigationbar
          history={
            Object {
              "action": "POP",
              "block": [Function],
              "canGo": [Function],
              "createHref": [Function],
              "entries": Array [
                Object {
                  "hash": "",
                  "key": "cpf0mn",
                  "pathname": "/",
                  "search": "",
                  "state": undefined,
                },
              ],
              "go": [Function],
              "goBack": [Function],
              "goForward": [Function],
              "index": 0,
              "length": 1,
              "listen": [Function],
              "location": Object {
                "hash": "",
                "key": "cpf0mn",
                "pathname": "/",
                "search": "",
                "state": undefined,
              },
              "push": [Function],
              "replace": [Function],
            }
          }
          location={
            Object {
              "hash": "",
              "key": "cpf0mn",
              "pathname": "/",
              "search": "",
              "state": undefined,
            }
          }
          match={
            Object {
              "isExact": true,
              "params": Object {},
              "path": "/",
              "url": "/",
            }
          }
          role="manager"
        >
          <Navbar
            bg="light"
            collapseOnSelect={false}
            expand="lg"
            variant="light"
          >
            <nav
              className="navbar navbar-expand-lg navbar-light bg-light"
            >
              <NavbarBrand
                onClick={[Function]}
              >
                <span
                  className="navbar-brand"
                  onClick={[Function]}
                >
                  Employee Management System
                </span>
              </NavbarBrand>
              <NavbarToggle
                aria-controls="basic-navbar-nav"
                label="Toggle navigation"
              >
                <button
                  aria-controls="basic-navbar-nav"
                  aria-label="Toggle navigation"
                  className="navbar-toggler collapsed"
                  onClick={[Function]}
                  type="button"
                >
                  <span
                    className="navbar-toggler-icon"
                  />
                </button>
              </NavbarToggle>
              <NavbarCollapse
                id="basic-navbar-nav"
              >
                <Collapse
                  appear={false}
                  dimension="height"
                  getDimensionValue={[Function]}
                  id="basic-navbar-nav"
                  in={false}
                  mountOnEnter={false}
                  timeout={300}
                  unmountOnExit={false}
                >
                  <Transition
                    addEndListener={[Function]}
                    appear={false}
                    aria-expanded={null}
                    enter={true}
                    exit={true}
                    id="basic-navbar-nav"
                    in={false}
                    mountOnEnter={false}
                    onEnter={[Function]}
                    onEntered={[Function]}
                    onEntering={[Function]}
                    onExit={[Function]}
                    onExited={[Function]}
                    onExiting={[Function]}
                    timeout={300}
                    unmountOnExit={false}
                  >
                    <div
                      aria-expanded={null}
                      className="navbar-collapse collapse"
                      id="basic-navbar-nav"
                    >
                      <Nav
                        className="mr-auto"
                        fill={false}
                        justify={false}
                      >
                        <ForwardRef
                          as="div"
                          className="mr-auto navbar-nav"
                          onSelect={[Function]}
                        >
                          <div
                            className="mr-auto navbar-nav"
                            onKeyDown={[Function]}
                          >
                            <BrowserRouter>
                              <Router
                                history={
                                  Object {
                                    "action": "POP",
                                    "block": [Function],
                                    "createHref": [Function],
                                    "go": [Function],
                                    "goBack": [Function],
                                    "goForward": [Function],
                                    "length": 1,
                                    "listen": [Function],
                                    "location": Object {
                                      "hash": "",
                                      "pathname": "blank",
                                      "search": "",
                                      "state": undefined,
                                    },
                                    "push": [Function],
                                    "replace": [Function],
                                  }
                                }
                              >
                                <Link
                                  onClick={[Function]}
                                  to="/createprofile"
                                >
                                  <LinkAnchor
                                    href="/createprofile"
                                    navigate={[Function]}
                                    onClick={[Function]}
                                  >
                                    <a
                                      href="/createprofile"
                                      onClick={[Function]}
                                    >
                                      Add Developer
                                    </a>
                                  </LinkAnchor>
                                </Link>
                                <Link
                                  onClick={[Function]}
                                  to="/alldevelopers"
                                >
                                  <LinkAnchor
                                    href="/alldevelopers"
                                    navigate={[Function]}
                                    onClick={[Function]}
                                  >
                                    <a
                                      href="/alldevelopers"
                                      onClick={[Function]}
                                    >
                                      All Developer
                                    </a>
                                  </LinkAnchor>
                                </Link>
                              </Router>
                            </BrowserRouter>
                          </div>
                        </ForwardRef>
                      </Nav>
                      <Nav
                        fill={false}
                        justify={false}
                      >
                        <ForwardRef
                          as="div"
                          className="navbar-nav"
                          onSelect={[Function]}
                        >
                          <div
                            className="navbar-nav"
                            onKeyDown={[Function]}
                          >
                            <Button
                              active={false}
                              disabled={false}
                              onClick={[Function]}
                              type="button"
                              variant="outline-info"
                            >
                              <button
                                className="btn btn-outline-info"
                                disabled={false}
                                onClick={[Function]}
                                type="button"
                              >
                                Logout
                              </button>
                            </Button>
                          </div>
                        </ForwardRef>
                      </Nav>
                    </div>
                  </Transition>
                </Collapse>
              </NavbarCollapse>
            </nav>
          </Navbar>
        </Navigationbar>
      </withRouter(Navigationbar)>
      <styled.div>
        <div
          className="sc-AxiKw gA-drMl"
        >
          <styled.h1>
            <h1
              className="sc-AxhCb jQvLNr"
            >
              \`Swapped History of (
              )\`
            </h1>
          </styled.h1>
        </div>
      </styled.div>
    </SwapShifts>
  </Router>
</MemoryRouter>
`;

exports[`<SwapShift /> Component mounting the component(for unauthorize user) 1`] = `
<MemoryRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "0bd3vv",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "0bd3vv",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <SwapShifts
      location={
        Object {
          "state": Object {
            "user": Object {
              "email": "",
            },
          },
        }
      }
      role="developer"
    >
      <NotFound
        role="developer"
      >
        <withRouter(Navigationbar)
          role="developer"
        >
          <Navigationbar
            history={
              Object {
                "action": "POP",
                "block": [Function],
                "canGo": [Function],
                "createHref": [Function],
                "entries": Array [
                  Object {
                    "hash": "",
                    "key": "0bd3vv",
                    "pathname": "/",
                    "search": "",
                    "state": undefined,
                  },
                ],
                "go": [Function],
                "goBack": [Function],
                "goForward": [Function],
                "index": 0,
                "length": 1,
                "listen": [Function],
                "location": Object {
                  "hash": "",
                  "key": "0bd3vv",
                  "pathname": "/",
                  "search": "",
                  "state": undefined,
                },
                "push": [Function],
                "replace": [Function],
              }
            }
            location={
              Object {
                "hash": "",
                "key": "0bd3vv",
                "pathname": "/",
                "search": "",
                "state": undefined,
              }
            }
            match={
              Object {
                "isExact": true,
                "params": Object {},
                "path": "/",
                "url": "/",
              }
            }
            role="developer"
          >
            <Navbar
              bg="light"
              collapseOnSelect={false}
              expand="lg"
              fluid={true}
              inverse={true}
              variant="light"
            >
              <nav
                className="navbar navbar-expand-lg navbar-light bg-light"
                fluid={true}
                inverse={true}
              >
                <NavbarBrand
                  onClick={[Function]}
                >
                  <span
                    className="navbar-brand"
                    onClick={[Function]}
                  >
                    Employee Management System
                  </span>
                </NavbarBrand>
                <NavbarToggle
                  aria-controls="basic-navbar-nav"
                  label="Toggle navigation"
                >
                  <button
                    aria-controls="basic-navbar-nav"
                    aria-label="Toggle navigation"
                    className="navbar-toggler collapsed"
                    onClick={[Function]}
                    type="button"
                  >
                    <span
                      className="navbar-toggler-icon"
                    />
                  </button>
                </NavbarToggle>
                <NavbarCollapse
                  id="basic-navbar-nav"
                >
                  <Collapse
                    appear={false}
                    dimension="height"
                    getDimensionValue={[Function]}
                    id="basic-navbar-nav"
                    in={false}
                    mountOnEnter={false}
                    timeout={300}
                    unmountOnExit={false}
                  >
                    <Transition
                      addEndListener={[Function]}
                      appear={false}
                      aria-expanded={null}
                      enter={true}
                      exit={true}
                      id="basic-navbar-nav"
                      in={false}
                      mountOnEnter={false}
                      onEnter={[Function]}
                      onEntered={[Function]}
                      onEntering={[Function]}
                      onExit={[Function]}
                      onExited={[Function]}
                      onExiting={[Function]}
                      timeout={300}
                      unmountOnExit={false}
                    >
                      <div
                        aria-expanded={null}
                        className="navbar-collapse collapse"
                        id="basic-navbar-nav"
                      >
                        <Nav
                          className="mr-auto"
                          fill={false}
                          justify={false}
                        >
                          <ForwardRef
                            as="div"
                            className="mr-auto navbar-nav"
                            onSelect={[Function]}
                          >
                            <div
                              className="mr-auto navbar-nav"
                              onKeyDown={[Function]}
                            >
                              <BrowserRouter>
                                <Router
                                  history={
                                    Object {
                                      "action": "POP",
                                      "block": [Function],
                                      "createHref": [Function],
                                      "go": [Function],
                                      "goBack": [Function],
                                      "goForward": [Function],
                                      "length": 1,
                                      "listen": [Function],
                                      "location": Object {
                                        "hash": "",
                                        "pathname": "blank",
                                        "search": "",
                                        "state": undefined,
                                      },
                                      "push": [Function],
                                      "replace": [Function],
                                    }
                                  }
                                >
                                  <Link
                                    onClick={[Function]}
                                    to="/tasks"
                                  >
                                    <LinkAnchor
                                      href="/tasks"
                                      navigate={[Function]}
                                      onClick={[Function]}
                                    >
                                      <a
                                        href="/tasks"
                                        onClick={[Function]}
                                      >
                                        Tasks
                                      </a>
                                    </LinkAnchor>
                                  </Link>
                                  <Link
                                    onClick={[Function]}
                                    to="/timings"
                                  >
                                    <LinkAnchor
                                      href="/timings"
                                      navigate={[Function]}
                                      onClick={[Function]}
                                    >
                                      <a
                                        href="/timings"
                                        onClick={[Function]}
                                      >
                                        Timings
                                      </a>
                                    </LinkAnchor>
                                  </Link>
                                  <Link
                                    onClick={[Function]}
                                    to="/swapshift"
                                  >
                                    <LinkAnchor
                                      href="/swapshift"
                                      navigate={[Function]}
                                      onClick={[Function]}
                                    >
                                      <a
                                        href="/swapshift"
                                        onClick={[Function]}
                                      >
                                        Swap Requests
                                      </a>
                                    </LinkAnchor>
                                  </Link>
                                </Router>
                              </BrowserRouter>
                            </div>
                          </ForwardRef>
                        </Nav>
                        <Nav
                          fill={false}
                          justify={false}
                        >
                          <ForwardRef
                            as="div"
                            className="navbar-nav"
                            onSelect={[Function]}
                          >
                            <div
                              className="navbar-nav"
                              onKeyDown={[Function]}
                            >
                              <Button
                                active={false}
                                disabled={false}
                                onClick={[Function]}
                                type="button"
                                variant="outline-info"
                              >
                                <button
                                  className="btn btn-outline-info"
                                  disabled={false}
                                  onClick={[Function]}
                                  type="button"
                                >
                                  Logout
                                </button>
                              </Button>
                            </div>
                          </ForwardRef>
                        </Nav>
                      </div>
                    </Transition>
                  </Collapse>
                </NavbarCollapse>
              </nav>
            </Navbar>
          </Navigationbar>
        </withRouter(Navigationbar)>
        <styled.div>
          <div
            className="sc-AxjAm bcThxe"
          >
            <styled.h1>
              <h1
                className="sc-AxirZ kHuMvb"
              >
                Not Found...
              </h1>
            </styled.h1>
          </div>
        </styled.div>
      </NotFound>
    </SwapShifts>
  </Router>
</MemoryRouter>
`;
